jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -2524,7 +2524,101 @@ public boolean startsWith(String prefix) {
 	public boolean startsWith(String prefix, int start) {
 		return regionMatches(start, prefix, 0, prefix.lengthInternal());
 	}
+	
+/*[IF Java11]*/
+	/**
+	 * Strip leading and trailing white spaces from a string.
+	 * 
+	 * @return a substring of the original containing no leading
+	 * or trailing white spaces.
+	 */
+	public String strip() {
+		return stripSub(true, true);
+	}
+	
+	/**
+	 * Strip leading white spaces from a string.
+	 * 
+	 * @return a substring of the original containing no leading
+	 * white spaces.
+	 */
+	public String stripLeading() {
+		return stripSub(true, false);
+	}
+	
+	/**
+	 * Strip trailing white spaces from a string.
+	 * 
+	 * @return a substring of the original containing no trailing
+	 * white spaces.
+	 */
+	public String stripTrailing() {
+		return stripSub(false, true);
+	}
+	
+	/**
+	 * Strip leading and/or trailing white spaces from a string
+	 * 
+	 * @param set to true if leading white spaces should be removed,
+	 * otherwise leading spaces are ignored.
+	 * @param set to true if trailing white spaces should be removed,
+	 * otherwise trailing spaces are ignored.
+	 * 
+	 * @return a substring of the original containing no leading
+	 * or trailing white spaces (depending on parameters).
+	 */
+	private String stripSub(boolean stripLeading, boolean stripTrailing) {
+		int start = 0;
+		int last = lengthInternal() - 1;
+		int end = last;
+
+		value.getClass(); // Implicit null check
 
+		// Check if the String is compressed
+		if (enableCompression &&  (null == compressionFlag || coder == LATIN1)) {
+			if (stripLeading) {
+				while ((start <= end) 
+					&& Character.isWhitespace(helpers.byteToCharUnsigned(helpers.getByteFromArrayByIndex(value, start)))
+				) {
+					start++;
+				}
+			}
+
+			if (stripTrailing) {
+				while ((end >= start) 
+					&& Character.isWhitespace(helpers.byteToCharUnsigned(helpers.getByteFromArrayByIndex(value, end)))
+				) {
+					end--;
+				}
+			}
+			
+			if (start == 0 && end == last) {
+				return this;
+			} else {
+				return new String(value, start, end - start + 1, true);
+			}
+		} else {
+			if (stripLeading) {
+				while ((start <= end) && Character.isWhitespace(charAtInternal(start))) {
+					start++;
+				}
+			}
+
+			if (stripTrailing) {
+				while ((end >= start) && Character.isWhitespace(charAtInternal(end))) {
+					end--;
+				}
+			}
+
+			if (start == 0 && end == last) {
+				return this;
+			} else {
+				return new String(value, start, end - start + 1, false);
```

Comment:
If the result is empty this can return "" instead of creating a new String object.